src/EA.Weee.Web/Areas/Admin/Mappings/ToViewModel/UploadObligationsViewModelMap.cs

```diff
@@ -41,17 +45,22 @@ private void SetErrors(UploadObligationsViewModelMapTransfer source, UploadOblig
             var displayDataError = false;
             var displayFileError = false;
 
-            if (source.ErrorData != null)
+            if (source.DisplayNotification)
             {
-                displayDataError = source.ErrorData.Any(r => dataErrorTypes.Contains(r.ErrorType));
-                displayFileError =
-                    source.ErrorData.Any(r => r.ErrorType == SchemeObligationUploadErrorType.File);
-                model.NumberOfDataErrors = source.ErrorData.Count(r => dataErrorTypes.Contains(r.ErrorType));
-                model.DisplaySuccessMessage = displayDataError == false && displayFileError == false;
-            }
+                if (source.ErrorData != null)
+                {
+                    displayDataError = source.ErrorData.Any(r => dataErrorTypes.Contains(r.ErrorType));
+                    displayFileError =
+                        source.ErrorData.Any(r => r.ErrorType == SchemeObligationUploadErrorType.File);
+                    model.NumberOfDataErrors = source.ErrorData.Count(r => dataErrorTypes.Contains(r.ErrorType));
+                    model.DisplaySuccessMessage = displayDataError == false && displayFileError == false && source.DisplayNotification;
+                }
 
-            model.DisplayFormatError = displayFileError;
-            model.DisplayDataError = displayFileError != true && displayDataError;
+                model.UploadedMessage =
+                    $"You have successfully uploaded the obligations for the compliance year {source.SelectedComplianceYear}";
+                model.DisplayFormatError = displayFileError && source.DisplayNotification;
```

Comment:
is that not going to be true only if there is error date? suggesting need to be in the if (source.ErrorData != null)